tests/update/updates_packagekit_kde.pm

```diff
@@ -63,6 +63,19 @@ sub run {
                     last;
                 }
             }
+            # Check, if there are more updates available
```

Comment:
I don't understand why this should be necessary. The while loop comment in L37 already mentions that this should be covered ("First update package manager, …") and also the corresponding test module "updates_packagekit_gpk" does not seem to need your change. So I am not sure why would need this. Best also take a look in https://bugzilla.suse.com/show_bug.cgi?id=992773

Reply:
Yes, it is necessary, because nothing triggers another installation round of pending packages see https://openqa.opensuse.org/tests/755521#step/updates_packagekit_kde/21. It installs only zypper update, then it fails. I reproduced behavior in virtual machine and human interaction to install pending updates is necessary.

Comment on line 37 seems to be misleading.

updates_packagekit_gpk is not relevant to this issue, it is for gnome, it has completely different design, it even triggers reboot during update if needed. Look on https://openqa.opensuse.org/tests/755518#step/updates_packagekit_gpk/16 it install zypper update first,  reboot system https://openqa.opensuse.org/tests/755518#step/updates_packagekit_gpk/19 , install rest of the packages https://openqa.opensuse.org/tests/755518#step/updates_packagekit_gpk/33.

Mentioned bug bsc#992773 is one of those "who cares", created in 2016 closed in 2018 as wontfix, just because Leap 42.1 ended its life. Moreover this bug is similart/same like bsc#1041112, about delayed status of updater widget and it is handled after lines 51.

What I see updates_packagekit_kde is not optimal, and should be probably re-done a bit. It soft-fails at situations which are reported as bugs, but those "probably" will stay untouched.  I just want to fix failing scenario (which blocks openSUSE updates) at this point.

If you think that my assumptions are not right, I will welcome any feedback which will solve it.